app/controllers/v3/service_instances_controller.rb

```diff
@@ -23,6 +23,7 @@
 require 'decorators/field_service_instance_offering_decorator'
 require 'decorators/field_service_instance_broker_decorator'
 require 'controllers/v3/mixins/service_permissions'
+require 'controllers/services/lifecycle/service_instance_purger'
```

Comment:
Hum... I see your point about this being in the `controllers` directory rather than in the `actions`.

A thing to notice is that there's nothing "v2" specific in the [purger](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/services/lifecycle/service_instance_purger.rb) code. I don't see how we would rewrite it differently than the current implementation.

I can duplicate that code and add it to the `actions` folder so it follows the v3 architecture 🤔

Reply:
I would be fine with duplicating to make it available for v3. 
Or- if there is any way to move the file to be more v3-like and make v2 depend on the moved file?

Reply:
I thought about that and I'm kind of more inclined to not change v2 to use the "v3" file. I think making v3 depend on a file that's used on v2 makes us cautious when changing it (like the broker client) but we often disregard the v2 implementation when changing files marked as "v3". I also looked at the other controllers and I cannot find any v2 code using v3 actions, so might be a good idea not to start it now 😛